src/libraries/System.Reflection.Metadata/tests/Metadata/Ecma335/Encoding/InstructionEncoderTests.cs

```diff
@@ -475,6 +475,35 @@ public void Branch()
             }, builder.ToArray());
         }
 
+        [Fact]
+        public void Switch()
+        {
+            var builder = new BlobBuilder();
+            var controlFlowBuilder = new ControlFlowBuilder();
+            var il = new InstructionEncoder(builder, controlFlowBuilder);
+            var l = il.DefineLabel();
+            var switchEncoder = il.Switch(4);
+            Assert.Throws<InvalidOperationException>(() => il.OpCode(ILOpCode.Nop));
+            Assert.Throws<InvalidOperationException>(() => il.Token(0));
+            Assert.Throws<InvalidOperationException>(() => il.MarkLabel(l));
+            Assert.Throws<InvalidOperationException>(() => controlFlowBuilder.AddFinallyRegion(l, l, l, l));
+            Assert.Throws<InvalidOperationException>(() => new MethodBodyStreamEncoder(new BlobBuilder()).AddMethodBody(il));
+            switchEncoder.Branch(l);
+            switchEncoder.Branch(l);
+            switchEncoder.Branch(l);
+            switchEncoder.Branch(l);
+            Assert.Throws<InvalidOperationException>(() => switchEncoder.Branch(l));
```

Comment:
Are we missing a test for the too-few branches case?

Reply:
https://github.com/dotnet/runtime/blob/395b8e5bb9a2d1014420ff8d7b0002e45fa28cff/src/libraries/System.Reflection.Metadata/tests/Metadata/Ecma335/Encoding/InstructionEncoderTests.cs#L485-L490

I added some asserts just after calling `Switch`, if zero branches can be considered too few.